Framework/Core/test/test_HistogramRegistry.cxx

```diff
@@ -19,18 +19,20 @@ using namespace o2::framework;
 
 HistogramRegistry foo()
 {
-  return {"r", true, {{"histo", "histo", {"TH1F", 100, 0, 1}}}};
+  return HistogramRegistry{"r", true, {histSpec<TH1F>("histo", "histo", "TH1F", {{100, 0, 1}})}};
 }
 
 BOOST_AUTO_TEST_CASE(HistogramRegistryLookup)
 {
   /// Construct a registry object with direct declaration
-  HistogramRegistry registry{"registry", true, {{"eta", "#Eta", {"TH1F", 100, -2.0, 2.0}}, {"phi", "#Phi", {"TH1D", 102, 0, 2 * M_PI}}, {"pt", "p_{T}", {"TH1D", 1002, -0.01, 50.1}}}};
+  HistogramRegistry registry{"registry", true, {histSpec<TH1F>("eta", "#Eta", "TH1F", {{100, -2.0, 2.0}}), histSpec<TH1D>("phi", "#Phi", "TH1D", {{102, 0, 2 * M_PI}}), histSpec<TH1D>("pt", "p_{T}", "TH1D", {{1002, -0.01, 50.1}}), histSpec<TH2F>("ptToPt", "ptToPt", "TH2F", {{100, -0.01, 10.01}, {100, -0.01, 10.01}})}};
```

Comment:
Why do you you need to pass the type and the string with the type name?

Reply:
The string is a remainder from the primary version, if it didn't have any purpose, I will delete it.

Reply:
My point is that if you have `histSpec<TH1F>()` you know you need to put `"TH1F"` in the spec. Unless that serves some other usecase.

Reply:
To be honest, I don't know if the spec string is needed now?
Hm, for that I'd have to store hardcoded lists of types and strings to set string without many ifs etc.
Actually, then one could pass just string and no templates will be needed. If hardcoded list is not a problem, maybe it would be better from the user's point of view...?